libsoftwarecontainer/unit-test/CMakeLists.txt

```diff
@@ -31,17 +31,28 @@ set(TEST_LIBRARY_DEPENDENCIES
 )
 include_directories(${LIBSOFTWARECONTAINER_DIR}/include)
 
+set(GATEWAY_TEST_FILES "")
+macro(add_gateway_test GATEWAY_FLAG FILENAME)
+    if (${GATEWAY_FLAG})
+        LIST(APPEND GATEWAY_TEST_FILES ${FILENAME})
+    endif()
+endmacro()
+
+add_gateway_test(ENABLE_CGROUPSGATEWAY cgroupsgateway_unittest.cpp)
+add_gateway_test(ENABLE_DBUSGATEWAY dbusgateway_unittest.cpp)
+add_gateway_test(ENABLE_DEVICENODEGATEWAY devicenodegateway_unittest.cpp)
+add_gateway_test(ENABLE_NETWORKGATEWAY networkgateway_unittest.cpp)
+add_gateway_test(ENABLE_NETWORKGATEWAY netlink_unittest.cpp)
+
+set(DEFAULT_GATEWAYS TRUE)
+add_gateway_test(DEFAULT_GATEWAYS envgateway_unittest.cpp)
+add_gateway_test(DEFAULT_GATEWAYS filegateway_unittest.cpp)
```

Comment:
Actually, it follows the pattern used for adding gateways in libsoftwarecontainer/src/CMakeLists.txt. It could be changed of course.